R/extremes-data.R

```diff
@@ -240,16 +240,16 @@ createDataRows <-
           if(!isUpchain){
             #Keep Maximum or Minimum based on current param
             if(param == "max"){
-              duplicateRows <- dataRows[order(dataRows$date, dataRows$related, dataRows$time, decreasing = TRUE),]
+              duplicateRows <- dataRows %>% arrange(desc(related)) %>% group_by(related) %>% arrange(date, time) %>% ungroup() %>% as.data.frame()
             } else  if (param == "min"){
-              duplicateRows <- dataRows[order(dataRows$date, dataRows$related, dataRows$time, decreasing = FALSE),]
+              duplicateRows <- dataRows %>% arrange(related) %>% group_by(related) %>% arrange(date, time) %>% ungroup() %>% as.data.frame()
             }
           } else {
             #Keep Maximum or Minimum based on current param
             if(param == "max"){
-              duplicateRows <- dataRows[order(dataRows$date, dataRows$primary, dataRows$time, decreasing = TRUE),]
+              duplicateRows <- dataRows %>% arrange(desc(primary)) %>% group_by(primary) %>% arrange(date, time) %>% ungroup() %>% as.data.frame()
             } else  if (param == "min"){
-              duplicateRows <- dataRows[order(dataRows$date, dataRows$primary, dataRows$time, decreasing = FALSE),]
+              duplicateRows <- dataRows %>% arrange(primary) %>% group_by(primary) %>% arrange(date, time) %>% ungroup() %>% as.data.frame()
```

Comment:
The reasoning behind using arrange before the group by is because I want the group with the highest (or lowest) value for its parameter (primary or related) to be at the top of the resulting data frame. Is that done by the group_by implicitly?